modules/efk-logging-collector-fluentd-v-rsyslog.adoc

```diff
@@ -0,0 +1,36 @@
+// Module included in the following assemblies:
+//
+// * logging/efk-logging-fluentd.adoc
+
+[id="efk-logging-collector-fluentd-v-rsyslog_{context}"]
+= About Fluentd and Rsyslog
+
+You can choose between Fluentd and Rsyslog as the logging collector. 
+
+Some of the noteable differences between the two collectors are:
+
+* No throttling in Rsyslog. With Fluentd, an administrator can reduce the rate at which the logs are read in by Fluentd before being processed. This feature is not currently available with Rsyslog.
+
+* No _built-in_ support for remote syslog forwarding in rsyslog. With Fluentd, you can use the `secure-forward` plug-in to send a copy of its logs to an external log aggregator. This feature is not currently available with Rsyslog.
++
+For Rsyslog, you can edit the Rsyslog configmap to add support for Syslog log forwarding using the *omfwd* module, see link:https://www.rsyslog.com/doc/v8-stable/configuration/modules/omfwd.html[]. To send logs to a different Rsyslog instance, you can the *omrelp* module, see link:https://www.rsyslog.com/doc/v8-stable/configuration/modules/omrelp.html[]
+
+* The Event Router is not supported for the Rsyslog log collector.
```

Comment:
I thought there is no relationship between event router and log collectors. Why do you document `The Event Router is not supported for the Rsyslog log collector.` here?
I've tested in my env before, and the event logs collected by event router all could be parsed correctly by rsyslog.

Reply:
Are event router records formatted in exactly the same way with fluentd as with rsyslog?  I think not.  I think that's what the problem is - rsyslog will collect those logs, but rsyslog will not format the logs in the same way as fluentd.

Reply:
I'm not sure. Since there is a bug about rsyslog parsing logs, I'm not able to check it now. I'll update my test result here once I could. Thank you.

Reply:
@mburke5678  Yes.

@richm  I compared the event logs collected by fluentd and rsyslog,  here is the logs: http://pastebin.test.redhat.com/786553.
If ignore the order of some objects, seems no difference.

Reply:
Then it looks like fluentd is broken - it doesn't seem to be processing the events correctly - it should "move" the events underneath the top level `kubernetes` field - https://github.com/ViaQ/fluent-plugin-viaq_data_model/blob/master/lib/fluent/plugin/filter_viaq_data_model.rb#L452 - `@process_kubernetes_events` is `true` by default.  Not sure what's going on here . . .

Reply:
Do you think this is a bug? I checked with 4.1 and 3.11:https://privatebin-it-iso.int.open.paas.redhat.com/?27170295f76a88ac#ccyZoBERqvkyPB+WZkPEBXbMKIgZLjwfYhb8aHt5y/g=, the event logs collected by fluentd are parsed to the same format as it in 4.2.

Reply:
Yes - it is a bug